File: BubbleSort/BubbleSort/BubbleSort.c
#include "D:/Coding/Projects/OtherProjects/TheSortingProject/Functions/C/MainMethod.h"

void bubbleSort(int arr[], int n)
{
  int i, j;
  for (i = 0; i < n - 1; i++)
    for (j = 0; j < n - i - 1; j++)
      if (arr[j] > arr[j + 1])
        swap(&arr[j], &arr[j + 1]);
}

int main(int argc, char const *argv[])
{
  int size = 4096;
  int *array = randomArray(size);
  normalArray(array, size);
  clock_t startTime = clock();
  bubbleSort(array, size);
  clock_t endTime = clock();
  sortedArray(array, size);
  timeTaken(startTime, endTime);
  return 0;
}

File: BubbleSort/BubbleSort/BubbleSort.java
package BubbleSort;

import Java.MainMethod;

class BubbleSortAlgorithm {
  public static void bubbleSort(int arr[]) {
    int n = arr.length;
    for (int i = 0; i < n; i++)
      for (int j = 0; j < n - i - 1; j++)
        if (arr[j] > arr[j + 1])
          MainMethod.swap(arr, j, j + 1);
  }
}

public class BubbleSort {
  public static void main(String[] args) {
    int[] array = MainMethod.randomArray();
    MainMethod.normalArray(array);

    long startTime = System.currentTimeMillis();
    BubbleSortAlgorithm.bubbleSort(array);
    long endTime = System.currentTimeMillis();

    MainMethod.sortedArray(array);
    MainMethod.timeTaken(startTime, endTime);
  }
}